server/raft.go

```diff
@@ -1290,21 +1289,42 @@ func (n *raft) StepDown(preferred ...string) error {
 			preferred = nil
 		}
 	}
+	// Can't pick ourselves.
+	if maybeLeader != noLeader && maybeLeader == n.id {
```

Comment:
Not a big deal but I guess this could just be `if maybeLeader == n.id`

Reply:
Will update. Thanks.